### PR TITLE
Fix PhpDoc: getByField methods can return null

### PR DESCRIPTION
**Pull request by blankse**
`$test = DataObject\Test::getByTest('Test', null, 1)` can return null.
PhpDoc is wrong and PhpStan not happy if I use it in an if :)

### CODE REVIEW

File: models/DataObject/ClassDefinition.php [hunk 1]
```diff
@@ -509,7 +509,7 @@ public function generateClassFiles($generateDefinitionFile = true)
                             $this->getName()
                         ).'\Listing|\\Pimcore\\Model\\DataObject\\'.ucfirst(
                             $this->getName()
-                        ).' getBy'.ucfirst(
+                        ).'|null getBy'.ucfirst(
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition.php:512]: Migration to regenerate class definitions?

Comment (dvesh3) [reply models/DataObject/ClassDefinition.php:512]: 🏓 @NiklasBr

Comment (blankse) [reply models/DataObject/ClassDefinition.php:512]: @dvesh3 I added a migration

**Comment by dvesh3** [comment]
@blankse thanks a lot!

**Pull request merged**